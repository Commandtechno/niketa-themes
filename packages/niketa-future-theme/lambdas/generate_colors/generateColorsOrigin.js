import * as chromatism from 'chromatism'
import {
  map,
  piped,
  random,
  shuffle,
  sortBy,
  split,
  splitEvery,
  take,
  tap,
  uniq,
} from 'rambdax'

import { getCombinations } from '../best_triangle/permutation'
import colorsOrigin from './colorsOrigin.json'
const MAX_COLORS = 400

export function findMiddleColor(colorTuple){
  if (colorTuple.length !== 2)
    throw new Error('It needs a tuple of colors')

  return chromatism.mid(colorTuple[ 0 ], colorTuple[ 1 ]).hex
}

export function sortResults(color){
  return chromatism.temperature(color)
}

export function generateColorsOrigin(seed){
  const colorSlices = splitEvery(seed, shuffle(colorsOrigin))
  const colorSliceIndex = random(0, colorSlices.length - 1)
  const currentSlice = colorSlices[ colorSliceIndex ]
  const allCombinations = []
  getCombinations(currentSlice, 2, allCombinations)

  const sortedResult = piped(
    allCombinations,
    map(split(' ')),
    map(findMiddleColor),
    tap(x => console.log(x.length, 'base')),
    uniq,
    sortBy(sortResults)
  )

  expect(take(MAX_COLORS, sortedResult)).toMatchSnapshot()
}
