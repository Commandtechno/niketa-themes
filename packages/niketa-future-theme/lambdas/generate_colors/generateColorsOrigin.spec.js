import { findMiddleColor, generateColorsOrigin } from './generateColorsOrigin.js'

test('happy', () => {
  console.time('generateColors')
  generateColorsOrigin(70)
  console.timeEnd('generateColors')
})

test.skip('findMiddleColor', () => {
  const result = findMiddleColor([ '#433daa', '#aadaa5' ])
  expect(result).toBe('#6cc7c5')
})
