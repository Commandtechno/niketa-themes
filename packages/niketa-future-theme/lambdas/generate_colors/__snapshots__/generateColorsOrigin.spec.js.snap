// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Array [
  "#4b75cf",
  "#b271e9",
  "#d930bf",
  "#443f7c",
  "#7079e0",
  "#a677ee",
  "#222e5a",
  "#0e315c",
  "#29396d",
  "#3c74d0",
  "#5c67c1",
  "#cc62e5",
  "#497adc",
  "#a076ec",
  "#c878ff",
  "#6e4a9c",
  "#0e91ff",
  "#6f7ce8",
  "#1c59a2",
  "#1c396d",
  "#2c366b",
  "#7b76e3",
  "#7283f6",
  "#d723b4",
  "#877df2",
  "#115ba7",
  "#263061",
  "#39407f",
  "#3c4283",
  "#0264b7",
  "#de58e5",
  "#9b77f1",
  "#356fce",
  "#8e6fe1",
  "#7677e8",
  "#dd23b7",
  "#212753",
  "#616fd7",
  "#963698",
  "#27376f",
  "#403470",
  "#293973",
  "#0e3970",
  "#484c99",
  "#4d3576",
  "#1e2855",
  "#7463ca",
  "#4f2f6e",
  "#5867cb",
  "#5975e3",
  "#724298",
  "#0d4181",
  "#443575",
  "#202959",
  "#1c478e",
  "#25448a",
  "#9c60d6",
  "#173b79",
  "#353675",
  "#946fed",
  "#d35fec",
  "#3866cc",
  "#003772",
  "#0a2e62",
  "#7970e9",
  "#5a6cdd",
  "#2855ad",
  "#2d50a5",
  "#5b72e9",
  "#9502f8",
  "#891eb3",
  "#a016f5",
  "#982aa6",
  "#a022db",
  "#890dd3",
  "#8c23a9",
  "#c24747",
  "#9f26c5",
  "#5f0694",
  "#6d05b8",
  "#b630d5",
  "#6904a7",
  "#7d16aa",
  "#8f21b0",
  "#8107de",
  "#a930ba",
  "#aa28d3",
  "#7400d3",
  "#b733c9",
  "#7e0bd6",
  "#8608ea",
  "#a82fbf",
  "#8900fc",
  "#a606ff",
  "#bd2ffd",
  "#971ace",
  "#a918ef",
  "#8f00dd",
  "#c24949",
  "#f66107",
  "#fa5f7b",
  "#f5567a",
  "#8619d2",
  "#8909c9",
  "#6805c6",
  "#f55e71",
  "#6007b3",
  "#860ef6",
  "#fa5785",
  "#7600b3",
  "#dc5715",
  "#8306bf",
  "#dd4d24",
  "#812929",
  "#c7443b",
  "#a526fd",
  "#e94ae4",
  "#b73e24",
  "#811bcb",
  "#711a9e",
  "#9d25b8",
  "#fa5685",
  "#a52ead",
  "#6f01f1",
  "#6806d7",
  "#5d09b6",
  "#9a2bb8",
  "#a32be0",
  "#f26233",
  "#f057b0",
  "#761cad",
  "#6f0fde",
  "#aa31cc",
  "#8921d4",
  "#851edb",
  "#c938d7",
  "#7c18e7",
  "#731bb0",
  "#6b11d6",
  "#67198d",
  "#5d0dc0",
  "#690bf4",
  "#ff6948",
  "#4601c0",
  "#5401e8",
  "#781adf",
  "#9c2cce",
  "#5207cb",
  "#ad0deb",
  "#8120d4",
  "#891da0",
  "#db4848",
  "#8f29ab",
  "#6515d0",
  "#880bb5",
  "#a12fd1",
  "#3e06b3",
  "#3a07a0",
  "#8223dd",
  "#5f16be",
  "#6517e0",
  "#8024a6",
  "#ca4c87",
  "#3d07c4",
  "#6117c9",
  "#8225a4",
  "#f5501c",
  "#8726de",
  "#762d2d",
  "#d85c12",
  "#7e23ed",
  "#962cee",
  "#782e2e",
  "#6219d9",
  "#3c09ff",
  "#9b2ef1",
  "#7e24b9",
  "#a231c2",
  "#5c18b6",
  "#8326f0",
  "#912bb9",
  "#ac00e6",
  "#e64917",
  "#4c15d9",
  "#9a2fd3",
  "#cb36d9",
  "#952db9",
  "#a4312c",
  "#ac1bd2",
  "#af1bd6",
  "#f74e5c",
  "#521adf",
  "#8829b3",
  "#651fcd",
  "#842aeb",
  "#e148e3",
  "#591ee1",
  "#bf531e",
  "#6322f0",
  "#c02cd7",
  "#3018f4",
  "#681fa4",
  "#9430ea",
  "#962fc7",
  "#7c28d5",
  "#8607aa",
  "#fa4e90",
  "#5921e9",
  "#461acb",
  "#7c2ae6",
  "#872bc7",
  "#a133c8",
  "#d35d37",
  "#8c2dcb",
  "#8729a1",
  "#441ddb",
  "#9b30ae",
  "#f858d5",
  "#6c23b3",
  "#3719c2",
  "#7625a9",
  "#892fe3",
  "#6525d5",
  "#6529f6",
  "#431bb9",
  "#2513a1",
  "#772494",
  "#2b1085",
  "#7124a5",
  "#7528c3",
  "#862aa8",
  "#1617bb",
  "#130f8c",
  "#9819b0",
  "#82289b",
  "#a638e0",
  "#ed4667",
  "#822aae",
  "#6823a8",
  "#3e29ff",
  "#bd365e",
  "#b23bd9",
  "#d66052",
  "#601f93",
  "#d93f6b",
  "#181dc9",
  "#9a17b3",
  "#a618c2",
  "#d85890",
  "#ab37c1",
  "#612ceb",
  "#e76863",
  "#182afe",
  "#4f1b8a",
  "#49166f",
  "#a735b4",
  "#a73ef7",
  "#8e31c1",
  "#7c31d9",
  "#9c4542",
  "#914130",
  "#ff718f",
  "#7934eb",
  "#ff486e",
  "#f74566",
  "#5d30e5",
  "#934331",
  "#5e32ea",
  "#9734c1",
  "#4530eb",
  "#6927a7",
  "#121ea9",
  "#fa4571",
  "#12188f",
  "#cf21ea",
  "#d15791",
  "#b52f51",
  "#be2f10",
  "#692dc1",
  "#841592",
  "#7629a0",
  "#992424",
  "#812eb0",
  "#a32a5b",
  "#161a8e",
  "#9214a4",
  "#7f36d9",
  "#cf6728",
  "#7c2dad",
  "#ed3f72",
  "#7539eb",
  "#2923ab",
  "#f464b9",
  "#812fb1",
  "#1a2fd8",
  "#f868b8",
  "#5c269e",
  "#4425a5",
  "#c443db",
  "#341979",
  "#8432ba",
  "#b713cf",
  "#4d2ab4",
  "#7e3be4",
  "#672799",
  "#501f82",
  "#c31cd8",
  "#bf2b25",
  "#6b2ba7",
  "#5232ce",
  "#6b299e",
  "#a24e48",
  "#482ab1",
  "#a04d47",
  "#923cda",
  "#3d1f89",
  "#d83775",
  "#211876",
  "#e93748",
  "#2925a4",
  "#702dab",
  "#f26ca6",
  "#f33e7e",
  "#6e299a",
  "#cf5f83",
  "#593ff5",
  "#503def",
  "#4432c9",
  "#c73066",
  "#802d9f",
  "#ff3f7a",
  "#fc8733",
  "#2131c9",
  "#1539e5",
  "#b65f34",
  "#b15b3f",
  "#d32c21",
  "#8b40e4",
  "#3b31c2",
  "#db2d06",
  "#e840a9",
  "#b25c41",
  "#7b2c9b",
  "#ef3552",
  "#b42a63",
  "#fa3c7d",
  "#f67f77",
  "#ec8315",
  "#081157",
  "#f73764",
  "#693ddc",
  "#c22e71",
  "#e07951",
  "#9233a9",
  "#f88d1d",
  "#a210b0",
  "#553fe1",
  "#f93030",
  "#343fe4",
  "#ad4cfa",
  "#bd29c3",
  "#983fd2",
  "#c414d5",
  "#dd3cdc",
  "#d93dd6",
  "#f18a1d",
  "#0b1c76",
  "#612f2f",
  "#ea2b1e",
  "#e14be3",
  "#de7f28",
  "#5e2689",
  "#623030",
  "#0f2796",
  "#e1803d",
  "#d17640",
  "#d55ca8",
  "#ef8a3e",
  "#7246ec",
  "#63288d",
  "#e58532",
  "#2939ca",
  "#af18ba",
  "#f22f5a",
  "#b850ff",
  "#e051ca",
  "#7d49f3",
  "#cd28d5",
  "#422ea5",
  "#ed2828",
  "#ec2e5e",
  "#8231a3",
  "#1a2faa",
  "#9f47e5",
  "#ef2e5f",
  "#2a2384",
  "#7240d6",
  "#8f49ee",
  "#3531ad",
  "#453bca",
]
`;
